inference-engine/tools/compile_tool/main.cpp

```diff
@@ -216,6 +239,31 @@ static InferenceEngine::Precision getOutputPrecision(const std::string &value) {
     return getPrecision(value, supported_precisions, "for output layer");
 }
 
+static InferenceEngine::Layout getLayout(const std::string &value) {
+    static const supported_layouts_t supported_layouts = {
+            { "NCHW", InferenceEngine::Layout::NCHW },
+            { "NHWC", InferenceEngine::Layout::NHWC },
+            { "NC", InferenceEngine::Layout::NC },
+            { "C", InferenceEngine::Layout::C }
+    };
+    return getLayout(value, supported_layouts);
+}
+
+static bool isMatchLayoutToDims(const InferenceEngine::Layout& layout, const size_t dimension) {
+    static const matchLayoutToDims_t matchLayoutToDims = {
+            {static_cast<size_t>(InferenceEngine::Layout::NCHW), 4 },
+            {static_cast<size_t>(InferenceEngine::Layout::NHWC), 4 },
+            {static_cast<size_t>(InferenceEngine::Layout::NC), 2 },
+            {static_cast<size_t>(InferenceEngine::Layout::C), 1 }};
```

Comment:
Do we care about CHW?

Reply:
If yes, I'd propose adding 5D layouts as well.

Reply:
FYI, it seems like one of target networks wants 3D input.

Reply:
Added 3D layout.